test/integration/device_multicast.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+"""Test multicast device."""
+
+import os
+import platform
+import socket
+import time
+import json
+import subprocess as subp
+
+from testlib import check, util
+from testlib.log import log
+from testlib.proc import Tinc, Script
+from testlib.test import Test
+
+
+def have_multicast() -> bool:
+    """Can we run multicast tests?"""
+
+    if platform.system() != "Linux":
+        return False
```

Comment:
Hm, multicast should work everywhere. If you want to test if the operating system it's running on support multicast, maybe just create two multicast enabled sockets in Python and check if they can send/receive to each other?

Reply:
Thanks, that was pretty stupid of me. As to why this check is needed at all, Windows doesn't support multicast without (it seems) additional configuration. I've also run into this on Linux (can't remember the exact setup, probably a `dummy` device with broadcast disabled and `lo` down). So it is possible.

Reply:
You're right, if we can't guarantee the network setup (it could even be firewall rules blocking multicast packets), then this test might be problematic. Maybe it should only be run in a network namespace then...

Reply:
But it looks good enough.